bwa/src/game.cpp

```diff
@@ -1,32 +1,47 @@
+#include <exception>
 #include <string>
 #include <utility>
 #include "game.hpp"
 
 constexpr const char* WINDOW_TITLE = "Bubble Warrior Adventures!";
 
 bwa::Game::Game() {
+	// Loads the config into the sol::state
 	_lua.script_file("config.lua");
+	// Converts the config's x and y resolution into a table
 	sol::table resolution = _lua["resolution"];
-	_font.loadFromFile(_lua["game_fonts"]["normal"]);
+	// Checks for font file and sets _font to it
+	if (!_font.loadFromFile(_lua["game_fonts"]["normal"]))
+	{
+		throw std::invalid_argument(_FILE_NOT_FOUND_ERROR + _lua["game_fonts"]["normal"].get<std::string>());
+	}
+	// Setting the look of the FPS counter
 	_text.setFont(_font);
 	_text.setFillColor(sf::Color::Yellow);
 	_text.setString("FPS:");
+	// pull the x and y window resolution coordinates from the config
 	auto xy = std::make_pair(
 		resolution["x"].get<unsigned>(), 
 		resolution["y"].get<unsigned>());
-	if (_lua["fullscreen"].get<bool>())
+	// Create a fullscreen window if set to true in config
+	if (_lua["fullscreen"].get<bool>()) {
 		_window.create({ xy.first, xy.second }, WINDOW_TITLE,
 			sf::Style::Fullscreen);
-	else
-		_window.create({ xy.first, xy.second }, WINDOW_TITLE);
+	}
+	// Create a normal window if fullscreen is set to false in config
+	else {
+	_window.create({ xy.first, xy.second }, WINDOW_TITLE);
 	_window.setVerticalSyncEnabled(true);
 	_gui.setWindow(_window);
+	}
```

Comment:
The following lines should be outside of the else clause because they should be called regardless of whether the window is fullscreen or not. It most likely worked for you, however, because you ran the code with `fullscreen` set to `false` but set to `true` would produce a runtime error. 

```cpp
_window.setVerticalSyncEnabled(true);
_gui.setWindow(_window);
```

Refer to above comments on omitted closing curly braces for if statements containing a single statement